sphinx/writers/html5.py

```diff
@@ -510,6 +510,20 @@ def depart_title(self, node: Element) -> None:
 
         super().depart_title(node)
 
+    # overwritten
+    def visit_rubric(self, node: Element) -> None:
+        if level := node.get("level"):
+            self.body.append(self.starttag(node, f'h{level}', '', CLASS='rubric'))
+        else:
+            super().visit_rubric(node)
+
+    # overwritten
+    def depart_rubric(self, node: Element) -> None:
+        if level := node.get("level"):
+            self.body.append(f'</h{level}>\n')
+        else:
```

Comment:
Add an assert on the allowed level (to avoid people injecting invalid attributes). And make sure that it's also of type int.

The level should not be at most 6 since otherwise it's not a valid HTML tag.

Reply:
a warning for unsupported level values has been added

Reply:
Say ``node['level'] = 9``, the warning would be emitted on ``visit_`` with no opening tag, but the closing `</h9>` tag would still be added. This isn't possible using the directive but we should guard against it (and test appropriately).